rustls/src/verify.rs

```diff
@@ -590,13 +590,24 @@ impl ClientCertVerifier for AllowAnyAuthenticatedClient {
         now: SystemTime,
     ) -> Result<ClientCertVerified, Error> {
         let (cert, chain, trustroots) = prepare(end_entity, intermediates, &self.roots)?;
+        let crls = self
+            .crls
+            .iter()
+            .map(|crl_der| webpki::CertRevocationList::from_der(crl_der))
+            .collect::<Vec<Result<CertRevocationList, webpki::Error>>>()
+            .into_iter()
+            .collect::<Result<Vec<CertRevocationList>, webpki::Error>>()
+            .map_err(|_| Error::InvalidCrl)?;
+        let crls = crls
+            .iter()
+            .collect::<Vec<&CertRevocationList>>();
         let now = webpki::Time::try_from(now).map_err(|_| Error::FailedToGetCurrentTime)?;
         cert.verify_is_valid_tls_client_cert(
             SUPPORTED_SIG_ALGS,
             &webpki::TlsClientTrustAnchors(&trustroots),
             &chain,
             now,
-            &[], // TODO(@cpu): parse CRL DER and provide.
+            crls.as_slice(),
```

Comment:
Maybe `verify_is_valid_tls_client_cert()` should take an `impl Iterator<Item = &CertRevocationList<'_>>` instead of a slice? Could avoid the intermediate allocations here.

Reply:
perhaps something stronger than just `impl Iterator` would be needed, as the CRLs need to be iterated a bunch of times per `verify_is_valid_tls_client_cert`?

Reply:
Ahh yes, we'll need to loop over it once per certificate considered for chain building. I suppose in that case a slice would be best after all.

Reply:
For my own education what do you mean by "something stronger"? Is the appeal of maintaining the slice related to some overhead of an `impl Iterator` approach? I have no strong pref. here, just want to understand the trade-offs better.

Reply:
I don't know how to write it precisely, but it'd be something like `impl Iterator<...> + Clone` (ie, it's an Iterator, but you can also clone the iterator to get more) or a function that returns a fresh `impl Iterator` each time.

Reply:
I haven't tackled this feedback yet (want to try some other stuff first) but it's on my radar.